core/network/PacketBuilder.hpp

```diff
@@ -865,4 +865,32 @@ struct PacketBuilder {
         return {};
       return packet;
     }
+
+    /**
+     * @brief Constructs a PingPacket with the specified timestamp.
+     *
+     * @param timestamp Timestamp to include in the ping packet.
+     * @return PingPacket Packet with header.type set to Ping, header.size set to the packet size, and timestamp set to the provided value.
+     */
+    static PingPacket makePing(std::uint32_t timestamp) {
+      PingPacket packet{};
+      packet.header.type = PacketType::Ping;
+      packet.header.size = sizeof(packet);
+      packet.timestamp = timestamp;
+      return packet;
+    }
+
+    /**
+     * @brief Constructs a PongPacket with the specified timestamp.
+     *
+     * @param timestamp Timestamp to include in the pong packet.
+     * @return PongPacket Packet with header.type set to Pong, header.size set to the packet size, and timestamp set to the provided value.
+     */
+    static PongPacket makePong(std::uint32_t timestamp) {
+      PongPacket packet{};
+      packet.header.type = PacketType::Pong;
+      packet.header.size = sizeof(packet);
+      packet.timestamp = timestamp;
+      return packet;
+    }
```

Comment:
utilise la nouvelle méthode pour calculer le `header.size`

```suggestion
    static PongPacket makePong(std::uint32_t timestamp) {
      PongPacket packet{};
      packet.header.type = PacketType::Pong;
      packet.timestamp = timestamp;

      if (!setPayloadSizeFromSerialization(packet, "makePong"))
        return {};
      return packet;
    }
```